Target Java 11

lkiesow opened the pull request:
This patch sets the target Java version for most modules to 11 which is the
effective minimum requirement for Opencast 10 anyway. This allows the usage of
newer Java syntax and features (Example incuded).

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

lkiesow committed:
Target Java 11

This patch sets the target Java version for most modules to 11 which is the
effective minimum requirement for Opencast 10 anyway. This allows the usage of
newer Java syntax and features (Example incuded).

lkiesow committed:
Update OSGi Bundler

This patch updates the OSGi bundler to be compatible with Java 10 and
above. It also fixes a number of dependency wiring or declaration
problems.

LukasKalbertodt reviewed (approved):
Very nice! I was looking forward to that. 

It compiles fine for me, tests are green and it seems to work. The patch also looks good to me. I don't quite understand the reason for those `Import-Package` changes, but oh well. It works :P

lkiesow commented:
> I don't quite understand the reason for those Import-Package changes, but oh well. It works :P

The new bundler, probably in combination with the Java 10 (11?) changes to the internal Java modules seems to cause the imports to work slightly different so that more requirements end up in the manifests. There aren't available at runtime which causes OSGi to complain (wiring errors). These additional libraries shouldn't be necessary (it did run just fine without them before after all), so I adjusted the imports. That, and I fixed a few actual errors which seemed to have worked by luck before.

gregorydlogan commented on modules/security-aai/pom.xml:
```diff
@@ -13,6 +13,7 @@
   <properties>
     <opencast.basedir>${project.basedir}/../..</opencast.basedir>
     <checkstyle.skip>false</checkstyle.skip>
+    <java.release>8</java.release>
```
What is the AAI code doing that restricts it to JDK 8?

lkiesow replied:
Test failures based on the Spring test library it is using.
